packages/chain-agnostic-permission/src/adapters/caip-permission-adapter-session-scopes.ts

```diff
@@ -126,3 +127,33 @@ export const getSessionScopes = (
     }),
   );
 };
+
+/**
+ * Get the permitted accounts for a given scope.
+ *
+ * @param caip25CaveatValue - The CAIP-25 CaveatValue to get the permitted accounts for
+ * @param scopes - The scopes to get the permitted accounts for
+ * @returns An array of permitted accounts
+ */
+export const getPermittedAccountsForScopes = (
+  caip25CaveatValue: Pick<
+    Caip25CaveatValue,
+    'requiredScopes' | 'optionalScopes'
+  >,
+  scopes: CaipChainId[],
+): CaipAccountId[] => {
+  const scopeAccounts: CaipAccountId[] = [];
+
+  scopes.forEach((scope) => {
+    const requiredScope = caip25CaveatValue.requiredScopes[scope];
+    const optionalScope = caip25CaveatValue.optionalScopes[scope];
+    if (requiredScope) {
+      scopeAccounts.push(...requiredScope.accounts);
+    }
+
+    if (optionalScope) {
+      scopeAccounts.push(...optionalScope.accounts);
+    }
+  });
+  return scopeAccounts;
```

Comment:
may need a uniq/dedupe here

Reply:
probably shouldn't be necessary but can add!

Reply:
since these are CAIP-10 addresses they really shouldn't be duplicated right?

Reply:
i mean that the same exact CAIP-10 address may exist in both the required and optional scopes and i'm not certain it makes sense for that address to appear twice in the return result of this helper

Reply:
> I mean that the same exact CAIP-10 address may exist in both the required and optional scopes and i'm not certain it makes sense for that address to appear twice in the return result of this helper

Ah I forgot that was possible 🤔 Well I'll dedupe it just in case then

Reply:
https://github.com/MetaMask/core/pull/5491/commits/89b51ab5fa67b59bbb3d9a05a4a0d5d1ad46c7da